scripts/globals/pirates.lua

```diff
@@ -0,0 +1,267 @@
+-----------------------------------
+-- Pirates helpers
+-----------------------------------
+require("scripts/globals/pathfind")
+
+xi = xi or {}
+xi.pirates = xi.pirates or {}
+xi.pirates.status =
+{
+    IDLE       = 0,
+    PREPARING  = 1,
+    ARRIVING   = 2,
+    SPAWNING   = 3,
+    ATTACKING  = 4,
+    DESPAWNING = 5,
+    DEPARTING  = 6,
+    FINISHED   = 7
+}
+
+local function SetZoneMusic(zone, block, track)
+    for _, player in pairs(zone:getPlayers()) do
+        player:ChangeMusic(block, track)
+    end
+end
+
+xi.pirates.init = function(ID)
+    local ship = GetNPCByID(ID.npc.PIRATE_SHIP.id)
+    ship:setAnimPath(ID.npc.PIRATE_SHIP.anim_path)
+    ship:setStatus(xi.status.DISAPPEAR)
+
+    for k, pirate in pairs(ID.npc.PIRATES) do
+        local npc = GetNPCByID(k)
+        npc:setPos(pirate.start_pos)
+        npc:initNpcAi()
+    end
+end
+
+xi.pirates.start = function(ID)
+    local ship = GetNPCByID(ID.npc.PIRATE_SHIP.id)
+    xi.pirates.setStatus(ship, xi.pirates.status.IDLE)
+    xi.pirates.despawnMobs(ID)
+    xi.pirates.despawnNPCs(ID)
+    xi.pirates.reset(ID)
+end
```

Comment:
Use the code in the code below to match the updated xi.pirates.reset(zone) function.
```
xi.pirates.start = function(zone)
     local ship = GetNPCByID(zones[zone:getID()].npc.PIRATE_SHIP.id)
     ship:setLocalVar(xi.pirates.statusvar, xi.pirates.status.IDLE)
     xi.pirates.reset(zone)
end

Reply:
See above